src/Observable.ts

```diff
@@ -247,7 +247,7 @@ export class Observable<T> implements CoreOperators<T>  {
   publishBehavior: (value: any) => ConnectableObservable<T>;
   publishReplay: (bufferSize?: number, windowTime?: number, scheduler?: Scheduler) => ConnectableObservable<T>;
   publishLast: () => ConnectableObservable<T>;
-  race: (...observables: Array<Observable<any>>) => Observable<any>;
+  race: (...observables: Array<Observable<T>>) => Observable<T>;
```

Comment:
Fundamentally it is implemented, it's just not obvious because [`subscribeToResult`](https://github.com/ReactiveX/RxJS/blob/master/src/util/subscribeToResult.ts) consumes Arrays, Iterables, Observabes and Promises.